### PR TITLE
fix: The cors middleware didn't work

### PR DESCRIPTION
**Pull request by hwbrzzl**
Closes # 

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

**Review by devhaozi — APPROVED** [review]
LGTM

File: cors.go [hunk 1]
```diff
@@ -1,17 +1,47 @@
...
+
+		if !needCors {
+			ctx.Request().Next()
+			return
+		}
+
 		allowedMethods := ConfigFacade.Get("cors.allowed_methods").([]string)
 		if len(allowedMethods) == 1 && allowedMethods[0] == "*" {
-			allowedMethods = []string{nethttp.MethodPost, nethttp.MethodGet, nethttp.MethodOptions, nethttp.MethodPut, nethttp.MethodDelete}
+			allowedMethods = []string{http.MethodGet, http.MethodPost, http.MethodHead, http.MethodPut, http.MethodDelete, http.MethodPatch}
```
Comment (devhaozi) [inline cors.go:44]: `contracts/http`我记得也加了方法的映射，这样或许不用导入`net/http`。

File: cors.go [hunk 2]
```diff
@@ -26,9 +56,9 @@ func Cors() httpcontract.Middleware {
 
 		instance.HandlerFunc(ctx.Response().Writer(), ctx.Request().Origin())
 
-		if ctx.Request().Origin().Method == nethttp.MethodOptions &&
+		if ctx.Request().Origin().Method == http.MethodOptions &&
 			ctx.Request().Header("Access-Control-Request-Method") != "" {
-			ctx.Request().AbortWithStatus(nethttp.StatusNoContent)
+			ctx.Request().AbortWithStatus(http.StatusNoContent)
```
Comment (devhaozi) [inline cors.go:61]: 同上

File: cors.go (hunk 1 above)
Comment (hwbrzzl) [reply cors.go:44]: Make sense, I'll optimize it in another PR.

**Pull request merged**